### PR TITLE
style: inputNumber border unmerged

### LINKED ISSUE #44997: Space.Compact 数字输入框带有前/后置标签边框样式问题
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/space-compact-shu-zi-shu-ru-kuang-dai-you-qian-hou-zhi-biao-qian-bian-kuang-yang-shi-wen-ti-wfrkq8?file=/src/App.js)

### Steps to reproduce

无

### What is expected?

边框合并

### What is actually happening?

image

image

| Environment | Info |
| --- | --- |
| antd | 5.9.2 |
| React | 18.2.0 |
| System | - |
| Browser | - |

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/44997
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Resolve the border issue with inputNumber when using addonBefore in Space.Compact.         |
| 🇨🇳 Chinese | 解决 inputNumber 在 Space.Compact 中使用 addonBefore  时的边框问题          |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 46ec2cf</samp>

Fix input number and button group alignment in RTL mode. Adjust the CSS style of `components/input-number/style/index.ts` to apply a negative margin to the input number element.

### <samp>🤖 Generated by Copilot at 46ec2cf</samp>

* Fix input number and button group alignment in RTL mode by adding negative margin to the end of input number element ([link](https://github.com/ant-design/ant-design/pull/45004/files?diff=unified&w=0#diff-63d0bbc4ed8cb1745da77de552d75aa0fd581ef3eb2d25d0108b2d55eb5f1857R180))

### CODE REVIEW

File: components/input-number/style/index.ts [hunk 1]
```diff
@@ -177,6 +177,7 @@ const genInputNumberStyles: GenerateStyle<InputNumberToken> = (token: InputNumbe
             display: 'inline-block',
             textAlign: 'start',
             verticalAlign: 'top', // https://github.com/ant-design/ant-design/issues/6403
+            marginInlineEnd: -1, // https://github.com/ant-design/ant-design/issues/44997
```
Comment (caijf) [inline components/input-number/style/index.ts:180]: 要在 Space.Compact 包裹的时候加，而不是直接加吧？目前是加错位置了

Comment (Yuiai01) [reply components/input-number/style/index.ts]: done 了～

File: components/input-number/style/index.ts [hunk 2]
```diff
@@ -361,6 +362,15 @@ const genInputNumberStyles: GenerateStyle<InputNumberToken> = (token: InputNumbe
         },
       },
     },
+
+    // Space
+    {
+      [`${antCls}-space-compact`]: {
```
Comment (MadCcc) [inline components/input-number/style/index.ts:368]: 要判断下是不是最后一个

Comment (Yuiai01) [reply components/input-number/style/index.ts:368]: 好

**Comment by MadCcc** [comment]
snapshot 要更新下

**Comment by Yuiai01** [comment]
> snapshot 要更新下

done～

**Review by MadCcc — APPROVED** [review]

**Comment by AlanQtten** [comment]
官网的Space.Compact文档需要更新下，里面漏掉了InputNumber

**Comment by Yuiai01** [comment]
> 官网的Space.Compact文档需要更新下，里面漏掉了InputNumber

好～

**Pull request merged**